packages/api/src/external/carequality/document/process-document-retrieval-results.ts

```diff
@@ -0,0 +1,170 @@
+import { ingestIntoSearchEngine } from "../../aws/opensearch";
+
+export async function processDocumentRetrievalResults({
+  requestId,
+  patientId,
+  cxId,
+  documentRetrievalResults,
+}: {
+  requestId: string;
+  patientId: string;
+  cxId: string;
+  documentRetrievalResults: DocumentRetrievalResult[];
+}): Promise<void> {
+  try {
+    for (const result of documentRetrievalResults) {
+      const { operationOutcome } = result.data;
+      const issuesWithGateway = operationOutcome?.issue?.length ?? 0;
+      const successDocsCount = result.data.documentReference?.length ?? 0;
+
+      tallyDocQueryProgressWithSource({
+        patient: { id: patientId, cxId: cxId },
+        progress: {
+          successful: successDocsCount,
+          errors: issuesWithGateway,
+        },
+        type: "download",
+        requestId,
+        source: MedicalDataSource.CAREQUALITY,
+      });
+
+      await handleDocReferences(result.data.documentReference, requestId, patientId, cxId);
+    }
+
+    await setDocQueryProgressWithSource({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: {
+        status: "completed",
+      },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+  } catch (error) {
+    const msg = `Failed to process documents in Carequality.`;
+    console.log(`${msg}. Error: ${errorToString(error)}`);
+
+    await setDocQueryProgressWithSource({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: { status: "failed" },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+
+    capture.message(msg, {
+      extra: {
+        context: `cq.processingDocuments`,
+        error,
+        patientId: patientId,
+        requestId,
+        cxId,
+      },
+      level: "error",
+    });
+    throw error;
+  }
+}
+
+async function handleDocReferences(
+  docRefs: DocumentReference[],
+  requestId: string,
+  patientId: string,
+  cxId: string
+) {
+  let errorCountConvertible = 0;
+
+  const { log } = out(`CQ handleDocReferences - requestId ${requestId}, M patient ${patientId}`);
+
+  const currentFHIRDocRefs = await getDocuments({
+    cxId,
+    patientId,
+    documentIds: docRefs.map(doc => doc.metriportId ?? ""),
+  });
+
+  const transactionBundle: Bundle = {
+    resourceType: "Bundle",
+    type: "transaction",
+    entry: [],
+  };
+
+  for (const docRef of docRefs) {
+    const isDocConvertible = isConvertible(docRef.contentType);
+    const shouldConvert = isDocConvertible && docRef.isNew;
+
+    if (!docRef.isNew) {
+      errorCountConvertible++;
+    }
+
+    if (shouldConvert) {
+      try {
+        await convertCDAToFHIR({
+          patient: {
+            id: patientId,
+            cxId,
+          },
+          document: {
+            id: docRef.metriportId ?? "",
+            content: { mimeType: docRef.contentType ?? "" },
+          },
+          s3FileName: docRef.fileName ?? "",
+          s3BucketName: docRef.fileLocation ?? "",
```

Comment:
don't we absolutely need both of these fields? should we throw an error if either one doesn't exist, since the conversion won't be able to actually do anything?